pkg/terraform/gather/vsphere/ip.go

```diff
@@ -0,0 +1,117 @@
+// Package vsphere contains utilities that help gather vsphere specific
+// information from terraform state.
+package vsphere
+
+import (
+	"context"
+	"net/url"
+	"time"
+
+	installertypes "github.com/openshift/installer/pkg/types"
+	"github.com/pkg/errors"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/soap"
+	"github.com/vmware/govmomi/vim25/types"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+
+	"github.com/openshift/installer/pkg/terraform"
+)
+
+func configureVSphereClient(ctx context.Context, vcenter, username, password string) (*vim25.Client, error) {
```

Comment:
We are also doing this in destroy... and eventually for creating the VM Template. I wonder if we shouldn't put this in a more shareable location?
https://github.com/openshift/installer/pull/2893/files#diff-42147831e863a7799f598c4f7ced9dbdR56